substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/locks/SingleThreadedVMLockSupport.java

```diff
@@ -37,8 +38,16 @@
  * Support of {@link VMMutex} and {@link VMCondition} in single-threaded environments. No real
  * locking is necessary.
  */
-final class SingleThreadedVMLockSupport {
-    // Empty class to have the same name as the source file.
+final class SingleThreadedVMLockSupport extends VMLockSupport {
+    @Override
+    public VMMutex[] getMutexes() {
+        return null;
+    }
+
+    @Override
+    public VMCondition[] getConditions() {
+        return null;
+    }
```

Comment:
These should either return an empty array (with a check during printing to say something like `VM Mutexes: (none)`, or a comment on the superclass should say that `null` is a valid return value.